src/richie/apps/search/indexers/courses.py

```diff
@@ -184,8 +185,9 @@ def get_data_for_es(cls, index, action):
                 )
 
             course_runs = course.get_course_runs()
+            categories = course.get_categories()
```

Comment:
I think you should retrieve the related published `title` objects and the `public_extention_id` in the same query. It should look something like this:
```suggestion
from django.db.models import Prefetch
...
categories = course.get_categories().prefetch_related(
    Prefetch(
        "extended_object__title_set",
        to_attr="published_titles",
        queryset=Title.objects.filter(published=True).only("language", "title"),
    )
)
.only("public_extension", "published_titles")
.distinct()
```

See example usage here: https://github.com/openfun/richie/blob/master/src/richie/apps/search/indexers/categories.py#L61